cdap-common/src/main/java/co/cask/cdap/common/zookeeper/ZKExtOperations.java

```diff
@@ -148,7 +158,7 @@
   }
 
   /**
-   * Update the content of the given node. If the node doesn't exists, it will try to create the node. If the node
+   * Update the content of the given node. If the node doesn't exist, it will try to create the node. If the node
```

Comment:
Is it possible for this operation to go on indefinitely? Any reason why we don't provide maxRetry value? Since we don't do anything even if a cancel is called on the returned Future.